src/IECoreScene/SceneCache.cpp

```diff
@@ -1301,6 +1371,14 @@ class SceneCache::WriterImplementation : public SceneCache::Implementation
 			}
 		}
 
+		void writeSet(const Name& name, const IECore::PathMatcherData *set )
+		{
+			IndexedIOPtr setsIO = m_indexedIO->subdirectory( setsEntry, IndexedIO::CreateIfMissing );
+
+			const Object *o = set;
```

Comment:
This is annoying, isn't it? I think the problem is that the derived class `save( SaveContext *context ) const` protected method masks the base class `save( IndexedIOPtr ioInterface, const IndexedIO::EntryID &name )` public method. All my fault, and I've been meaning to fix it.

For what it's worth I tend to use `set->Object::save()` to do it in one line and hopefully make it a bit more obvious why the jiggery pokery is needed.